### PR TITLE
Integrate Lidar usage in cortex platform and add test platform

### PR DESCRIPTION
**Pull request by ecourtois**
Add a new obstacles module and remove old one.
Add an example using new obstacles module with Lidar driver.
Integrate Lidar in cortex platform and cup2019 application.
Remove usage of vl53l0x.
Add test board, platform and application.
Add sdcard usage example.

### CODE REVIEW

File: boards/cogip-nucleo-f446re/Kconfig [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2020 Inria
```
Comment (gdoffe) [inline boards/cogip-nucleo-f446re/Kconfig:1]: please change header for cogip one

File: boards/cogip-nucleo-f446re/doc.txt [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/**
+@defgroup    boards_nucleo-f446re STM32 Nucleo-F446RE
+@ingroup     boards_common_nucleo64
+@brief       Support for the STM32 Nucleo-F446RE
```
Comment (gdoffe) [inline boards/cogip-nucleo-f446re/doc.txt:4]: Maybe we should change this too to something custom. Have you try 'make doc' ?

File: examples/sdcard/include/sdcard_spi_params.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+static const sdcard_spi_params_t sdcard_spi_params[] = {
+    {
+        .spi_dev = SPI_DEV(0),
+        .cs = GPIO_PIN(PORT_A, 8),
+        // .clk            = GPIO_PIN(PORT_A, 5),
+        .clk = GPIO_PIN(PORT_B, 3),
+        // .mosi           = GPIO_PIN(PORT_A, 7),
+        .mosi = GPIO_PIN(PORT_B, 5),
+        // .miso           = GPIO_PIN(PORT_A, 6),
```
Comment (gdoffe) [inline examples/sdcard/include/sdcard_spi_params.h:17]: can you remove commented code ?

File: examples/sdcard/main.c [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        printf("Last errno: %d\n", errsv);
+        perror("Last error");
+        return 1;
+    }
+    printf("File '%s' close: success\n", test_write_filename);
+
+    return 0;
+}
+
+
```
Comment (gdoffe) [inline examples/sdcard/main.c:148]: you can remove one line

File: examples/sdcard/main.c [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+    char line_buf[SHELL_DEFAULT_BUFSIZE];
+    shell_run(shell_commands, line_buf, SHELL_DEFAULT_BUFSIZE);
+
+    // res = vfs_umount(&flash_mount);
+    // if (res < 0) {
+    //     printf("Error while unmounting %s\n", flash_mount.mount_point);
+    //     return 1;
+    // }
+    // printf("%s successfully unmounted\n", flash_mount.mount_point);
```
Comment (gdoffe) [inline examples/sdcard/main.c:226]: remove commented code

File: lib/cogip_defs/include/cogip_defs.h [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+static inline int pose_equal(const pose_t *p1, const pose_t *p2)
+{
+    if ((p1 != NULL) && (p2 != NULL)) {
+        return p1 == p2
+               || (p1->x == p2->x && p1->y == p2->y && p1->O == p2->O);
+    }
+    else {
+        return 0;
+    }
+}
```
Comment (gdoffe) [inline lib/cogip_defs/include/cogip_defs.h:38]: doxygen please

File: examples/lidar_obstacles/Makefile [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+USEMODULE += cogip_defs
+USEMODULE += lds01_dma
+USEMODULE += obstacles
+USEMODULE += shell_menu
+
+INCLUDES += -I$(CURDIR)/include
+
+# CFLAGS += -DCONFIG_SHELL_NO_ECHO
+# CFLAGS += -DCONFIG_SHELL_NO_PROMPT
```
Comment (gdoffe) [inline examples/lidar_obstacles/Makefile:19]: Please remove commented code

File: examples/lidar_obstacles/lidar_utils.c [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    return NULL;
+}
+
+/* Callback for LDS01 DMA driver */
+void new_frame_available_cb(void)
+{
+    event_post(&new_frame_queue, &new_frame_event);
+}
+
+/* print data in JSON format */
```
Comment (gdoffe) [inline examples/lidar_obstacles/lidar_utils.c:49]: Print

File: examples/shmem/lidar_obstacles.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+shm_data = ShmData.from_buffer(shm_ptr)
+
+for i in range(360):
+    shm_data.lidar[i] = 65535
+
+input(
+    f"""
+Shared memory key = {shm_ptr.key}
+
+Start the native application the gdb.
```
Comment (gdoffe) [inline examples/shmem/lidar_obstacles.py:26]: missing word ?

File: examples/shmem/lidar_obstacles.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+In the RIOT shell:
+    - use the key with '_set_shmem_key,
```
Comment (gdoffe) [inline examples/shmem/lidar_obstacles.py:28]: missing '

File: boards/cogip-nucleo-f446re/Kconfig (hunk 1 above)
Comment (ecourtois) [reply boards/cogip-nucleo-f446re/Kconfig]: Done

File: boards/cogip-nucleo-f446re/doc.txt (hunk 2 above)
Comment (ecourtois) [reply boards/cogip-nucleo-f446re/doc.txt]: Done.
For 'make doc', I think some 'doxygen.txt' are missing in 'applications', 'boards', 'examples', 'lib and 'sys', but not sure it should be part of this PR.

File: examples/sdcard/include/sdcard_spi_params.h (hunk 3 above)
Comment (ecourtois) [reply examples/sdcard/include/sdcard_spi_params.h]: Done

File: examples/sdcard/main.c (hunk 5 above)
Comment (ecourtois) [reply examples/sdcard/main.c]: Done

File: lib/cogip_defs/include/cogip_defs.h (hunk 6 above)
Comment (ecourtois) [reply lib/cogip_defs/include/cogip_defs.h:85]: Done

File: examples/lidar_obstacles/lidar_utils.c (hunk 8 above)
Comment (ecourtois) [reply examples/lidar_obstacles/lidar_utils.c]: Done

File: examples/shmem/lidar_obstacles.py (hunk 10 above)
Comment (ecourtois) [reply examples/shmem/lidar_obstacles.py]: Done.
File moved to 'examples/lidar_obstacles'.

File: examples/sdcard/main.c (hunk 4 above)
Comment (ecourtois) [reply examples/sdcard/main.c]: Done

File: examples/lidar_obstacles/Makefile (hunk 7 above)
Comment (ecourtois) [reply examples/lidar_obstacles/Makefile]: Done

File: examples/shmem/lidar_obstacles.py (hunk 9 above)
Comment (ecourtois) [reply examples/shmem/lidar_obstacles.py]: Done

File: boards/cogip-nucleo-f446re/doc.txt (hunk 2 above)
Comment (gdoffe) [reply boards/cogip-nucleo-f446re/doc.txt]: Yes I know there is so much documentation to still write. Thats a huge work.
What I mean is that it is a good practice to test 'make doc' on new/refactored code at least to check if sections are well ordered.

File: examples/lidar_obstacles/lidar_obstacles.c [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+            }
+
+            /* Don't exit the loop if only the 3 consecutive angles have no obstacle */
+            bool continue_loop = false;
+            for (uint32_t next = last + 1; next < last + 3; next++) {
+                if (raw_distances[next % LDS01_NB_ANGLES] < lidar_obstacles_params.max_distance) {
+                    continue_loop = true;
+                    break;
+                }
+            }
```
Comment (gdoffe) [inline examples/lidar_obstacles/lidar_obstacles.c:83]: Avoid hard coded values, use a #define

File: applications/app_test/path.c [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+//         },
+//         .allow_reverse = TRUE,
+//         .max_speed = MAX_SPEED,
+//         .act = NULL,
+//     },
+
+//     /*
+//      * End of path
+//      */
+// };
```
Comment (gdoffe) [inline applications/app_test/path.c:51]: remove commented code

File: examples/lidar_obstacles/lidar_obstacles.c (hunk 11 above)
Comment (ecourtois) [reply examples/lidar_obstacles/lidar_obstacles.c:86]: Done

File: applications/app_test/path.c (hunk 12 above)
Comment (ecourtois) [reply applications/app_test/path.c]: Done

**Comment by gdoffe** [comment]
LGTM

**Review by gdoffe — APPROVED** [review]

**Pull request merged**